enterprise/micronaut/src/org/netbeans/modules/micronaut/db/MicronautController.java

```diff
@@ -273,6 +206,86 @@ public void addChangeListener(ChangeListener l) {
     public void removeChangeListener(ChangeListener l) {
     }
 
+    @NbBundle.Messages({
+        "MSG_NoRepositories=No repository interface found in {0}",
+        "MSG_SelectRepository=Select Data Repository Interfaces",
+        "MSG_SelectRepository_Prompt=Repositories to be called from Controllers",
+        "MSG_SelectControllerName=Controller Name"
+    })
+    private static CreateFromTemplateHandler handler(boolean fromRepository) {
+        return new CreateFromTemplateHandler() {
+            @Override
+            protected boolean accept(CreateDescriptor desc) {
+                return true;
+            }
+
+            @Override
+            protected List<FileObject> createFromTemplate(CreateDescriptor desc) throws IOException {
+                try {
+                    final FileObject folder = desc.getTarget();
+                    final Project project = FileOwnerQuery.getOwner(folder);
+                    if (project == null) {
+                        DialogDisplayer.getDefault().notifyLater(new NotifyDescriptor.Message(Bundle.MSG_NoProject(folder.getPath()), NotifyDescriptor.ERROR_MESSAGE));
+                        return Collections.emptyList();
+                    }
+                    if (fromRepository) {
+                        final SourceGroup sourceGroup = SourceGroups.getFolderSourceGroup(ProjectUtils.getSources(project).getSourceGroups(JavaProjectConstants.SOURCES_TYPE_JAVA), folder);
+                        if (sourceGroup == null) {
+                            DialogDisplayer.getDefault().notifyLater(new NotifyDescriptor.Message(Bundle.MSG_NoSourceGroup(folder.getPath()), NotifyDescriptor.ERROR_MESSAGE));
+                            return Collections.emptyList();
+                        }
+                        Set<ElementHandle<TypeElement>> repositoryClasses = getRepositoryClasses(sourceGroup);
+                        if (repositoryClasses.isEmpty()) {
+                            DialogDisplayer.getDefault().notifyLater(new NotifyDescriptor.Message(Bundle.MSG_NoRepositories(sourceGroup.getRootFolder().getPath()), NotifyDescriptor.ERROR_MESSAGE));
+                            return Collections.emptyList();
+                        }
+                        List<NotifyDescriptor.QuickPick.Item> items = repositoryClasses.stream().map(handle -> {
+                            String fqn = handle.getQualifiedName();
+                            int idx = fqn.lastIndexOf('.');
```

Comment:
minor nicpick: will report inner classes as 'in fullPkgName.OuterClass'

Reply:
To reliably separate fully qualified class name to fullPkgName and possible Outer.Nested class name, given `ElementHandle` should be resolved into the corresponding `TypeElement`, which is (IMHO) not worth of the effort here. The UI repository class selector displays simple class name as the label, followed by a less prominent class location that can be either a package or an outer class.